Ado improvements (#1914)
dbt-labs/docs.getdbt.com: website/docs/docs/dbt-cloud/cloud-configuring-dbt-cloud/setup-azure.md

4. [Connect Azure DevOps to your new app](#connect-azure-devops-to-your-new-app).
5. [Add your Azure AD app to dbt Cloud](#add-your-azure-ad-app-to-dbt-cloud).

Once the the Azure AD app is added to dbt Cloud, an account admin must also connect a service user via OAuth, which will be used to power headless actions in dbt Cloud such as deployment runs and CI.
1. [Connecting a Service User](#connecting-a-service-user).

Once the Azure AD app is added to dbt Cloud and the service user is connected, then dbt Cloud developers can personally authenticate in dbt Cloud from Azure DevOps. For more on this, see [Authenticate with Azure DevOps](docs/dbt-cloud/cloud-configuring-dbt-cloud/authenticate-azure).

## Register an Azure AD app

1. Sign into you Azure portal and click **Azure Active Directory** under Azure services.
2. Select **App registrations** in the left panel.
3. Select **New registration**. The form for creating a new Active Directory app opens.
4. Provide a name for your app. We recommend using, "dbt Labs Azure DevOps App".
5. Select **Accounts in any organizational directory (Any Azure AD directory - Multitenant)** as the Supported Account Types. 

> Many customers ask why they need to select Multitenant instead of Single tenant, and they frequently get this step wrong. Microsoft considers Azure DevOps (fka Visual Studio) and Azure Active Directory as separate tenants, and in order for this Active Directory application to work properly, you must select Multitenant.

Reviewer: > (fka Visual Studio)

What is fka? Do you mean aka? If so, we should spell it out or say "also called Visual Studio."

Author: formerly known as 

https://docs.microsoft.com/en-us/azure/devops/server/tfs-is-now-azure-devops-server?view=azure-devops

2. Remove the **Microsoft Graph / User Read** permission.
3. Click **Add a permission**.
4. Select **Azure DevOps**.

> 5. Select the **user_impersonation** permission. This is the least scoped and only permission available for Azure DevOps.

Reviewer: @schottj We should explain "least scoped" a little. It makes me wonder _least scoped compared to what_?
```suggestion
5. Select the **user_impersonation** permission. This permission limits the access to what is absolutely needed to run the Azure DevOps app successfully.
```

Author: Maybe we should just say: "This is the only permission available for Azure DevOps."
 
 We had a customer ask if there was any less privileged permission. And the answer is "No, this is the only permission available."

## Connect Azure DevOps to your new app

> If your Azure DevOps account is already connected to your Active Directory, then you can skip this step. However, if you're just getting set up, connect Azure DevOps to the Active Directory App you just created:

Reviewer: ```suggestion
If you have already connected your Azure DevOps account to Active Directory, then you can proceed to [Connecting a service user](#connecting-a-service-user). However, if you're just getting set up, connect Azure DevOps to the Active Directory App you just created:
```

Your Azure AD app should now be added to your dbt Cloud Account. People on your team who want to develop in dbt Cloud's IDE can now personally [authorize Azure DevOps from their profiles](dbt-cloud/cloud-configuring-dbt-cloud/authenticate-azure).

## Connecting a service user

> Because Azure DevOps forces all authentication to be linked to a user's permissions, we recommend you create a "service user" in Azure DevOps whose permissions which will be used to power headless actions in dbt Cloud such as deployment runs and CI. A service user is a pseudo user set up in the same way an admin would set up a real user, but it's given permissions specifically scoped for service to service interactions. You should avoid linking authentication to a real Azure DevOps user because if this person leaves your organization, dbt Cloud will lose privileges to the dbt Azure DevOps repositories, causing production runs to fail.

Reviewer 2: maybe not the place for this, but I'm pretty sure "service principal" is the Azure-ier name than "service user"
https://docs.microsoft.com/en-us/azure/active-directory/fundamentals/service-accounts-introduction-azure#types-of-azure-active-directory-service-accounts

Reviewer: Oh interesting, @dataders! I suggested using both because I feel like "Service user" is industry standard, or at least more familiar.

Author: Let's keep as "service user". We're not actually using an Azure "service principal" in the way that Azure describes it. We try to make that explicit here: "A service user is a pseudo user set up in the same way an admin would set up a real user,"

> Because Azure DevOps forces all authentication to be linked to a user's permissions, we recommend you create a "service user" in Azure DevOps whose permissions which will be used to power headless actions in dbt Cloud such as deployment runs and CI. A service user is a pseudo user set up in the same way an admin would set up a real user, but it's given permissions specifically scoped for service to service interactions. You should avoid linking authentication to a real Azure DevOps user because if this person leaves your organization, dbt Cloud will lose privileges to the dbt Azure DevOps repositories, causing production runs to fail.

Reviewer: ```suggestion
Because Azure DevOps forces all authentication to be linked to a user's permissions, we recommend you create a "service user" or "service principal" in Azure DevOps whose permissions which will be used to power headless actions in dbt Cloud such as deployment runs and CI. A service user is a pseudo user set up in the same way an admin would set up a real user, but it's given permissions specifically scoped for service to service interactions. You should avoid linking authentication to a real Azure DevOps user because if this person leaves your organization, dbt Cloud will lose privileges to the dbt Azure DevOps repositories, causing production runs to fail.
```

### More on Service Users

> A service user account must have the following permissions for all Azure DevOps projects/repos you want accessible in dbt Cloud:

Reviewer 2: making clearer...
```suggestion
A service user account must have the following Azure DevOps permissions for all Azure DevOps projects and repos you want accessible in dbt Cloud. Read more about how dbt Cloud uses each permission in the following paragraphs. 
```

 - **Project Reader**
 - **ViewSubscriptions**
 - **EditSubscriptions**

>  - **DeleteSubscriptions** * 

Reviewer 2: any idea why these permissions are needed?

Reviewer: @dataders I see the reasons for these permissions below, so I added a sentence to your suggestion above!

 - **PullRequestContribute**
 - **GenericContribute**

\* Note: **DeleteSubscriptions** permission might be included in **EditSubscriptions** depending on your version of Azure.

Some of these permissions are only accessible via the Azure DevOps API, for which documentation can be found [here](https://docs.microsoft.com/en-us/azure/devops/organizations/security/namespace-reference?view=azure-devops). We’ve also detailed more information on Azure DevOps API usage below to help accelerate the set up. Alternatively, you can use the Azure DevOps UI to enable permissions, but you cannot get the least permissioned set.

> :::info  The service user must have these permissions before dbt Cloud Project set up

Reviewer: ```suggestion
:::info  Provide the service user with required permissions before setting up a dbt Cloud project
```

> This service user's permissions will also power which repos a team can select from during dbt project set up, so an Azure DevOps admin must grant at minimum Project Reader access to the service user before setting up a project in dbt Cloud.

Reviewer: ```suggestion
This service user's permissions will also power which repositories a team can select from during dbt project set up, so an Azure DevOps admin must grant at minimum Project Reader access to the service user before setting up a project in dbt Cloud.
```

:::

> <details>

Reviewer: @schottj I think this permission info would be a lot easier to read as a table. Let me know if you're OK with that and I will update!

Author: I tried this at first, but I couldn't get the table entries to show code snippets. If you figure it out, would love help re-arranging!

Reviewer: Just to follow up: I looked into making a table here and we'd need to use an HTML table. This is doable but will take longer. So let's ship this and follow up later when we have time.
